sklearn/discriminant_analysis.py

```diff
@@ -254,6 +255,22 @@ def __init__(self, solver='svd', shrinkage=None, priors=None,
         self.store_covariance = store_covariance  # used only in svd solver
         self.tol = tol  # used only in svd solver
 
+    @property
+    def covariance_(self):
+        if hasattr(self, '_covariance_') and not self.store_covariance:
+            warnings.warn("from version 0.21 'covariance_' will be stored "
+                          "only if 'store_covariance' is True",
+                          DeprecationWarning)
+        return self._covariance_
+
+    @covariance_.setter
+    def covariance_(self, value):
+        if hasattr(self, '_covariance_') and not self.store_covariance:
+            warnings.warn("from version 0.21 'covariance_' will be stored "
+                          "only if 'store_covariance' is True",
+                          DeprecationWarning)
```

Comment:
I think this should be `FutureWarning` rather than `DeprecationWarning`. This should solve the testing issues.